### DISCUSSION TITLE
TextLayer text smaller when zooming in using OrbitView?

### DISCUSSION THREAD

**Discussion by nilscb** [visgl/deck.gl#6808] [Q&A]
TextLayer text smaller when zooming in using OrbitView?

I am experimenting with using TextLayer for showing labels on an axis in 3D. 
But I observe sometimes the weird effect that the labels are getting very much smaller when I zoom in on them!?
I am setting the textlayer property "sizeUnits" to "pixels" and also "sizeMin/MaxPixels" to lets say 24.
Could this be a defect in TextLayer perhaps? It should at least respect the "sizeMinPixels" I think.
Anyone have any thoughts on this?

Nils Christian Bonnevie
![text-layer-zoom](https://user-images.githubusercontent.com/136738/162389741-83c0c3e5-891d-41a5-96c2-8ebdc3e6cf98.png)

**Comment by Pessimistress** [comment]
https://github.com/visgl/deck.gl/discussions/6696#discussioncomment-2269035

**Reply by nilscb** [reply]
Setting sizeUnits to "meters" as was done in the above comment does not help unfortunately. It also does not make sence to me. Setting sizeUnits to "pixels" should be sufficient and should make the text size constant regardless of zoom but it does not.  (exept for a small change in size due to perspective)

**Reply by Pessimistress** [reply]
Did you read the linked comment? Size unit is irrelevant. The key is to constrain the view state target in `onViewStateChange`.

**Reply by nilscb** [reply]
View state target is what the camera looks at right. That is it defines the camera direction. I am only zooming in on text and expect the text to stay the same size in pixels. Only zooming should not change camera direction. But ok maybe I have misunderstood something fundamentally here (that has happened before) I'll have another try..

**Reply by Pessimistress** [reply]
Zooming does not change the camera direction, but it does change the camera position. The `OrbitView` does not know the bounds of your data, so the target may end up drifting away from what you want to look at.

**Reply by nilscb** [reply]
You are right in that the target very slightly drifts a bit but not enough for the camera to visibly change direction. Also if I in onViewStateChange constrain the target to a dead fixed position the 3D text still gets smaller when I zoom in on it.

**Reply by Pessimistress** [reply]
Well all I can do is to speculate based on a screenshot. If you create a codepen it will be easier to understand your issue.

**Reply by nilscb** [reply]
Ok here is a small example with a few labels along the z axis. Notice that when you zoom closer to the labels on the negative z axis they actually becomes smaller.

https://codepen.io/nilscb/pen/mdpjayE?editors=0011
